tests/service-catalog/test/pkg/report/report.go

```diff
@@ -0,0 +1,258 @@
+package report
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions/printers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	scClient "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	hbClient "github.com/kyma-project/helm-broker/pkg/client/clientset/versioned"
+	mappingClient "github.com/kyma-project/kyma/components/application-broker/pkg/client/clientset/versioned"
+	appClient "github.com/kyma-project/kyma/components/application-operator/pkg/client/clientset/versioned"
+	bucClient "github.com/kyma-project/kyma/components/service-binding-usage-controller/pkg/client/clientset/versioned"
+)
+
+type Report struct {
+	test      *testing.T
+	logWriter TestLogWriter
+	cfg       *rest.Config
+	opt       []Option
+	printer   printers.ResourcePrinter
+}
+
+func NewReport(t *testing.T, config *rest.Config, options ...Option) *Report {
+	return &Report{
+		test:      t,
+		logWriter: TestLogWriter{testing: t},
+		cfg:       config,
+		opt:       options,
+	}
+}
+
+type Option func(r *Report, ns string)
```

Comment:
We can call that type `ReportOption`, `LoggingOption` or `PrinterOption`. We can discuss it and decide